stdlib/public/runtime/DynamicCast.cpp

```diff
@@ -868,22 +879,25 @@ tryCastToAnyHashable(
   
   switch (srcType->getKind()) {
   case MetadataKind::ForeignClass: // CF -> String
-  case MetadataKind::ObjCClassWrapper: { // Obj-C -> String
+  case MetadataKind::Existential: { // AnyObject can be an NSString
 #if SWIFT_OBJC_INTEROP
-    auto cls = srcType;
-    auto nsString = getNSStringMetadata();
-    do {
-      if (cls == nsString) {
-        hashableConformance = getNSStringHashableConformance();
-        break;
-      }
-      cls = _swift_class_getSuperclass(cls);
-    } while (cls != nullptr);
-    break;
-#else
+    auto existentialType = cast<ExistentialTypeMetadata>(srcType);
+    // TODO: also handle non-class existentials that could contain an NSString
+    if (existentialType->getRepresentation() == ExistentialTypeRepresentation::Class) {
+      auto val = reinterpret_cast<const ClassExistentialContainer *>(srcValue)->Value;
+      auto cls = swift_getObjectType((HeapObject *)val);
```

Comment:
The Metadata APIs are a twisty maze of passageways all alike, is there a better way of getting this info out?

Reply:
You really shouldn't need to handle existential unwrapping here; if you fail here, the source should get unwrapped by the outer logic in `tryCast()` and this will get repeated.   If that's not working, then there's something else amiss.

Reply:
Oh neat. Ok, I’ll try that

Reply:
Much better!